application/src/main/java/org/togetherjava/tjbot/commands/help/UserBannedDeleteRecentThreadsListener.java

```diff
@@ -40,11 +42,12 @@ public UserBannedDeleteRecentThreadsListener(Database database) {
 
     @Override
     public void onGuildBan(GuildBanEvent event) {
-        RestAction.allOf(getRecentHelpThreads(event.getUser()).stream()
-            .map(event.getJDA()::getThreadChannelById)
-            .filter(Objects::nonNull)
-            .map(ThreadChannel::delete)
-            .toList()).queue();
+        getRecentHelpThreads(event.getUser()).forEach(channelId -> event.getJDA()
+            .getThreadChannelById(channelId)
+            .delete()
+            .queue(onSuccess -> {
+            }, onFailure -> logger.warn("Failed to delete thread {} from banned user.", channelId,
```

Comment:
the naming is incorrect. `onFailure` sounds as if it would be used as high order object, such as `queue(onSuccess, onFailure)`, but u use them as lambdas, so they are the input object. in this case `onFailure` is actually the `failure`/exception itself.

so `.queue(any -> {}, failure -> logger.warn(..., failure));`
or `.queue(any -> {}, Foo::onFailure);` with a corresponding method
or `.queue(any -> {}, onFailure);` where you created a `Consumer<Throwable> onFailure = ...` beforehand